cmd/migrate/main.go

```diff
@@ -330,28 +329,19 @@ func (m *chunkMover) moveChunks(ctx context.Context, threadID int, syncRangeCh <
 					chunks := schemaGroups[i][j:k]
 					//log.Printf("%v Processing chunks %v-%v of %v\n", threadID, j, k, len(schemaGroups[i]))
 
-					keys := make([]string, 0, len(chunks))
 					chks := make([]chunk.Chunk, 0, len(chunks))
 
-					// FetchChunks requires chunks to be ordered by external key.
-					sort.Slice(chunks, func(x, y int) bool {
-						return m.schema.ExternalKey(chunks[x].ChunkRef) < m.schema.ExternalKey(chunks[y].ChunkRef)
-					})
-					for _, chk := range chunks {
-						key := m.schema.ExternalKey(chk.ChunkRef)
-						keys = append(keys, key)
-						chks = append(chks, chk)
-					}
-					finalChks, err := f.FetchChunks(m.ctx, chks, keys)
+					chks = append(chks, chunks...)
+
+					finalChks, err := f.FetchChunks(m.ctx, chks)
```

Comment:
Why not pass `chks` here?

Reply:
good question, looking at this code a bit I think there are other problems here though, it seems like we overwrite the finalChks slice when fetching errors.

There were some recent changes here, will follow up with this with the folks that made that change and address this in a second PR